File: baekjoon/1806.py
n,s=map(int,input().split())
arr=list(map(int,input().split()))

st=0
end=0
value=0
answer=int(1e9)
while True:
    if value>=s:
        #answer=min(answer,end-st)
        value-=arr[st]
        st+=1
    elif end==n:
        break
    else:
        value+=arr[end]
        end+=1
    #print(st,end,value)
    if value>=s:
        # print(end,st)
        answer=min(answer,end-st)

if answer==int(1e9):
    print(0)
else:
    print(answer)

File: java/src/baekjoon/_1806.java
package baekjoon;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class _1806 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int start = 0;
        int end = 0;
        int sum = 0;
        int answer = Integer.MAX_VALUE;

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        while (true) {
            if (sum > k) {
                sum -= arr[start];
                start += 1;
            } else if (end == n) {
                break;
            } else {
                sum += arr[end];
                end += 1;
            }

            if (sum >= k) {
                answer = Math.min(answer, end - start);
            }
        }

        if (answer == Integer.MAX_VALUE) {
            System.out.println(0);
        } else {
            System.out.println(answer);
        }
    }
}
